hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/FSHLog.java

```diff
@@ -595,8 +595,12 @@ public void run() {
           Throwable lastException = null;
           try {
             TraceUtil.addTimelineAnnotation("syncing writer");
+            long unSyncedFlushSeq = highestUnsyncedTxid;
```

Comment:
Ya this looks to be an issue. We update the highestSyncedTxid with currentSequence which was calculated in above steps.  By the time when we actually do the sync, it would have been changed. The moment we call a sync (by any syncer tread) we would sync all appends till then.
cc @saintstack